mlrun/run.py

```diff
@@ -1218,3 +1226,609 @@ def wait_for_runs_completion(runs: list, sleep=3, timeout=0, silent=False):
+            :param obj:            The object to log.
+            :param key:            The key of the artifact.
+            :param logging_kwargs: Additional keyword arguments to pass to the `context.log_artifact` method.
+            """
+            context.log_artifact(
+                **logging_kwargs, item=key, body=cloudpickle.dumps(obj), format="pkl"
+            )
+
+        @staticmethod
+        def log_plot(context: MLClientCtx, obj, key: str, logging_kwargs: dict):
+            """
+            Log an object as a plot. Currently, supporting plots produced by one the following modules: `matplotlib`,
+            `seaborn`, `plotly` and `bokeh`.
+
+            :param context:        The MLRun context to log with.
+            :param obj:            The plot to log.
+            :param key:            The key of the artifact.
+            :param logging_kwargs: Additional keyword arguments to pass to the `context.log_artifact`.
+
+            :raise MLRunInvalidArgumentError: If the object type is not supported (meaning the plot was not produced by
+                                              one of the supported modules).
+            """
+            # Create the plot artifact according to the module produced the object:
+            artifact = None
+
+            # `matplotlib` and `seaborn`:
+            try:
+                import matplotlib.pyplot as plt
+
+                from mlrun.artifacts import PlotArtifact
+
+                # Get the figure:
+                figure = None
+                if isinstance(obj, plt.Figure):
+                    figure = obj
+                elif isinstance(obj, plt.Axes):
+                    if hasattr(obj, "get_figure"):
+                        figure = obj.get_figure()
+                    elif hasattr(obj, "figure"):
+                        figure = obj.figure
+                    elif hasattr(obj, "fig"):
+                        figure = obj.fig
+
+                # Create the artifact:
+                if figure is not None:
+                    artifact = PlotArtifact(key=key, body=figure)
+            except ModuleNotFoundError:
+                pass
+
+            # `plotly`:
+            if artifact is None:
+                try:
+                    import plotly
+
+                    from mlrun.artifacts import PlotlyArtifact
+
+                    if isinstance(obj, plotly.graph_objs.Figure):
+                        artifact = PlotlyArtifact(key=key, figure=obj)
+                except ModuleNotFoundError:
+                    pass
+
+            # `bokeh`:
+            if artifact is None:
+                try:
+                    import bokeh.plotting as bokeh_plt
+
+                    from mlrun.artifacts import BokehArtifact
+
+                    if isinstance(obj, bokeh_plt.Figure):
+                        artifact = BokehArtifact(key=key, figure=obj)
+                except ModuleNotFoundError:
+                    pass
```

Comment:
can this be used in more places? Maybe we should move it to some helper function, let me know what you think.

Reply:
I thought of making `context.log_plot` and then I'll call it from here. For now, let's keep it here. Trust me that if I ever find myself needing it elsewhere I'll move it to a utils something.